### PR TITLE
Fixing issues with Staff Record Questions

### PR DESCRIPTION
**Pull request by ChazUK**
Fixing submission issues, error messages and retaining answers on conditional radios

### CODE REVIEW

File: src/app/features/workers/salary/salary.component.html [hunk 1]
```diff
@@ -35,29 +35,31 @@ <h1 class="govuk-heading-xl">Are they paid hourly or annually?</h1>
...
+          </span>
+          <span class="govuk-input__currency">
+            <input
+              class="govuk-input govuk-date-input__input govuk-input--width-4"
+              [class.govuk-input--error]="submitted && form.get('hourlyRate').invalid"
+              formControlName="hourlyRate"
+              id="hourlyRate"
+              name="hourlyRate"
+              type="number"
+              pattern="^([0-9]*[.])?[0-9]+$"
```
Comment (omarCreativeDev) [inline src/app/features/workers/salary/salary.component.html]: can this pattern also live in a constants file please?

File: src/app/features/workers/salary/salary.component.html [hunk 2]
```diff
@@ -101,16 +103,18 @@ <h1 class="govuk-heading-xl">Are they paid hourly or annually?</h1>
...
-          />
+          <span class="govuk-input__currency">
+            <input
+              class="govuk-input govuk-date-input__input govuk-input--width-5"
+              [class.govuk-input--error]="submitted && form.get('annualRate').invalid"
+              formControlName="annualRate"
+              id="annualRate"
+              name="annualRate"
+              type="number"
+              pattern="^[0-9]+$"
```
Comment (omarCreativeDev) [inline src/app/features/workers/salary/salary.component.html]: this one too please

File: src/app/features/workers/salary/salary.component.html (hunk 1 above)
Comment (ChazUK) [reply src/app/features/workers/salary/salary.component.html]: done

File: src/app/features/workers/salary/salary.component.html (hunk 2 above)
Comment (ChazUK) [reply src/app/features/workers/salary/salary.component.html]: done

**Comment by ChazUK** [comment]
[Accessibility version: Date error messages are all caps](https://trello.com/c/JBl6JpYW/571-accessibility-version-date-error-messages-are-all-caps)

**Pull request merged**